task_processing/plugins/kubernetes/task_config.py

```diff
@@ -0,0 +1,57 @@
+import re
+import secrets
+
+from pyrsistent import field
+from pyrsistent import PMap
+from pyrsistent import PVector
+
+from task_processing.interfaces.task_executor import DefaultTaskConfigInterface
+
+BASE_58 = '123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz'
+UUID_BYTES = 6
+MAX_POD_NAME_LENGTH = 253
+REGEX_MATCH = '[a-z0-9]([.-a-z0-9]*[a-z0-9])?'
+
+
+class KubernetesTaskConfig(DefaultTaskConfigInterface):
+    def __invariant__(conf):
+        return(
+            (
+                len(conf.pod_name) > MAX_POD_NAME_LENGTH,
+                (
+                    f'Invalid format for pod_name {conf.pod_name}. ',
+                    f'Pod name must have up to {MAX_POD_NAME_LENGTH}.'
+                )
+            ),
+            (
+                re.match(REGEX_MATCH, conf.pod_name),
+                (
+                    f'Invalid format for pod_name {conf.pod_name}. ',
+                    f'Must comply with Kubernetes pod naming standards.'
+                )
+            )
+        )
```

Comment:
super-nit: 
```suggestion
    def __invariant__(conf):
        return (
            (
                len(conf.pod_name) > MAX_POD_NAME_LENGTH,
                (
                    f'Invalid format for pod_name {conf.pod_name}. ',
                    f'Pod name must have up to {MAX_POD_NAME_LENGTH}.'
                )
            ),
            (
                re.match(REGEX_MATCH, conf.pod_name),
                (
                    f'Invalid format for pod_name {conf.pod_name}. ',
                    f'Must comply with Kubernetes pod naming standards.'
                )
            )
        )
        
```

Reply:
we should probably add `black` to this repo to autoformat code at some point, I guess :p